src/gf.c

```diff
@@ -749,6 +749,7 @@ static int check_ambiguous_visitor(jl_typemap_entry_t *oldentry, struct typemap_
     struct ambiguous_matches_env *closure = container_of(closure0, struct ambiguous_matches_env, match);
     if (oldentry == closure->newentry)
         return 0; // finished once it finds the method that was just inserted
+    // TODO: instead, maybe stop once we hit something newentry is definitely more specific than
```

Comment:
+1

alternatively, I've been wondering if we can use the work done here to create a fast-path for `invalidate_conflicting` where we only test lambdas that were derived from methods that `check_ambiguous` deemed to have an intersection